hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/node/TestSCMNodeManager.java

```diff
@@ -1572,6 +1572,83 @@ public void testScmStatsFromNodeReport()
     }
   }
 
+  private List<StorageReportProto> generateStorageReportProto(
+      int volumeCount, UUID dnId, long capacity, long used) {
+    List<StorageReportProto> reports = new ArrayList<>(volumeCount);
+    boolean failed = true;
+    for (int x = 0; x < volumeCount; x++) {
+      long free = capacity - used;
+      String storagePath = testDir.getAbsolutePath() + "/" + dnId;
+      reports.add(HddsTestUtils
+          .createStorageReport(dnId, storagePath, capacity,
+              used, free, null, failed));
+      failed = !failed;
+    }
+    return reports;
+  }
+
+  @Test
```

Comment:
We can avoid repetition by parameterising the test. It can have only 1 dn and the parameters can be the `capacity` and `used` from the `generateStorageReportProto` and the expected result.

Regarding parameterising the test, here is an example

https://github.com/apache/ozone/blob/master/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFsSnapshot.java#L208-L251

Reply:
Thank you for these suggestions.
I will update later.